### DISCUSSION TITLE
Converting Eclipse SUMO Scenario to Eclipse MOSAIC Scenario

### DISCUSSION THREAD

**Discussion by realmaxneu** [eclipse/mosaic#28] [Q&A] [answered]
Hello,

is it possible to convert an SUMO Scenario to an Eclipse MOSAIC scenario?
I am using the scenario-convert tool with the option `--sumo2db`, but apparently it doesn't work without an existing database file.
Would that be the correct way to do so or did I overlook something?

**Comment by kschrab** [comment, marked as answer]
This is possible, but only to a limited extend.

First of all, the `--sumo2db` should indeed work to import a SUMO network file and should also create a new database file. This might be a regression we would need to fix in scenario-convert until the next release. However, even then the naming format of the edges/connections in the database would not be compatible with MOSAIC, and some features would not be available.

As a second solution, one could use the `SumoScenarioAmbassador` instead. This method allows to integrate and execute already existing SUMO scenarios with MOSAIC. However, again various features would not be available, such as routing. Mapping applications is possible, but only to a limited extend as well. We have a tutorial on our website covering this method, which integrates the LuST scenario with MOSAIC: https://www.eclipse.org/mosaic/tutorials/lust/

Please note, both methods are still under development and might be improved with future releases.

A bit of background: The database file is required for navigation capabilities, i.e. that applications in the Application simulator are able to get information about their location and that routes can be calculated during simulation. When coming from OSM data, we introduced a custom format for naming the edges and nodes. This special format helps us to match the position of the vehicle onto the edge/connection objects found in the database. When using a custom SUMO network file, this special naming format is not necessarily given, thus it would not be not compatible with the format MOSAIC expects and routing would not work.

**Reply by kschrab** [reply]
In recent pull requests (#39 #44) we come a bit closer towards a real simple integration of already existing SUMO scenarios into MOSAIC. With this solution you can take any existing SUMO configuration and run it with MOSAIC, and you can also add new entities via the Mapping. We will add a new tutorial for this feature on our website soon!
